File: 0014-longest-common-prefix/0014-longest-common-prefix.js
var longestCommonPrefix = function(strs) {
    if (strs.length === 0) return "";

    var start = strs[0];
    var newstr = "";
    var j = 0;

    while (j < start.length) {
        var f = 0;
        for (let temp of strs) {
            if (temp[j] !== start[j]) {
                f = 1;  // mismatch found
                break;
            }
        }

        if (f === 1) break;  // stop adding chars if mismatch
        newstr += start[j];   // add current character
        j += 1;
    }

    return newstr;
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        s = strs[0]
        for i in range(1, len(strs)):
            j = 0
            while j < len(s) and j < len(strs[i]):
                if s[j] != strs[i][j]:
                    break
                j += 1
            s = s[:j]

        return s